shared/middleware/helpers/auth_helpers.rb

```diff
@@ -23,6 +23,17 @@ def admin?
   current_user && !!current_user[:admin]
 end
 
+# @param [String] permission - Name of the permission we're interested in
+# @returns [Boolean] true if the current user has the specified dashboard permission
+def has_permission?(permission)
+  return false unless current_user
+
+  @user_permissions ||= {}
```

Comment:
Is this intended to function as a cache? Do middleware variables reset (as instance variables in ActiveRecord methods are) on every request?